addons-dev/aldryn-addons/aldryn_addons/settings.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+import imp
+import os
+import shutil
+import uuid
+from aldryn_addons import utils
+from pprint import pformat
+from django.core.exceptions import ImproperlyConfigured
+
+
+def save_settings_dump(settings, path):
+    to_dump = {key: value for key, value in settings.items() if key == key.upper()}
+    with open(path, 'w+') as fobj:
+        fobj.write(
+            pformat(
+                to_dump,
+                indent=4,
+            )
+        )
+
+
+def count_str(number):
+    return '{0:05d}'.format(number)
+
+
+def load(settings):
+    settings['BASE_DIR'] = settings.get(
+        'BASE_DIR',
+        os.path.dirname(os.path.abspath(settings['__file__']))
+    )
+    settings['ADDONS_DIR'] = settings.get(
+        'ADDONS_DIR',
+        os.path.join(settings['BASE_DIR'], 'addons')
+    )
+    settings['ADDONS_DEV_DIR'] = settings.get(
+        'ADDONS_DEV_DIR',
+        os.path.join(settings['BASE_DIR'], 'addons-dev')
+    )
+    utils.mkdirs(settings['ADDONS_DIR'])
+    # TODO: .debug is not multi-process safe!
+    debug_path = os.path.join(settings['ADDONS_DIR'], '.debug')
+    shutil.rmtree(debug_path, ignore_errors=True)
+    utils.mkdirs(debug_path)
+
+    def dump(obj, count, name):
+        dump_name = '{}-{}.dump'.format(count_str(count), name)
+        dump_path = os.path.join(debug_path, dump_name)
+        save_settings_dump(obj, dump_path)
+        return count + 1
+
+    debug_count = 0
+    debug_count = dump(settings, debug_count, 'initial')
+    # load global defaults
+    from django.conf import global_settings
```

Comment:
This code is run before/while django settings are configured. So to avoid circular imports and loading anything from django that requires django to already be configured, I tend to make imports from django as lazy as possible. For this specific import it could be put at the top though.

I also kind of like it to be together with the code that actually uses it in this case though (cleaner diff and related code together). What do you thinkg @jqb ? put it at the top anyway?